pandas/tseries/tests/test_period.py

```diff
@@ -2537,6 +2540,16 @@ def test_searchsorted(self):
             ValueError, 'Different period frequency: H',
             lambda: pidx.searchsorted(pd.Period('2014-01-01', freq='H')))
 
+    def test_round_trip_pickle(self):
+
+        p = Period('2000Q1')
+        pickle_path = os.path.join(tm.get_data_path(),
+                                   'period.pickle')
+
+        with open(pickle_path, 'wb') as f: pickle.dump(p, f)
```

Comment:
2 things

- use ``self.round_trip_pickle`` to test the pickling in ``test_period`` instead of the ad-hoc code above.
- fix ``generate_legacy_storeage_files.py`` to test ``Period`` (and ``Timestamp``) as singular objects. If you are comfortable with this, gr8, if not just do the above and i'll fix the rest.